### PR TITLE
feat(release): Enable CI trigger for v2.3 branch

### PR DESCRIPTION
**Pull request by imjoey**
Signed-off-by: imjoey <majunjiev@gmail.com>

Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance

___
### New feature or improvement
- Describe the details and related test reports.

As the preparation of v2.3 release, we need to continuously backport hot bugfixs to `v2.3` branch, so enable CI process at v2.3 branch would help us keep the release more stable. 

Since v2.3 will not be a LTS version, which means there will no activities in v2.3 branch after release v2.3, so this PR is going to use the single `v2.3` for all branch configurations.

### CODE REVIEW

File: .github/workflows/backend-cli-test.yml [hunk 1]
```diff
@@ -3,10 +3,10 @@ name: Backend CLI Test
 on:
   push:
     branches:
-      - master
+      - v2.3
```
Comment (juzhiyuan) [inline .github/workflows/backend-cli-test.yml:6]: 2.x is acceptable :)

Comment (juzhiyuan) [reply .github/workflows/backend-cli-test.yml:6]: both the master and v2.x need to be checked IMO

**Review by juzhiyuan — CHANGES_REQUESTED** [review]

File: .github/workflows/backend-cli-test.yml (hunk 1 above)
Comment (imjoey) [reply .github/workflows/backend-cli-test.yml:6]: Thanks for reviewing. IMHO, CI for `v2.3` will no longer exist after version `2.3` is released, cause it is not a LTS version. And this change will also not be backported into master branch.  For example, for the release version `2.4`, we can simply use `v2.4` here. Meanwhile, the `master` is the only branch name needed to listed here in the `master` branch.

Once we have LTS versions, then we can keep `master` and all LTS version branches in the `master` branch. So What do you think? 😄 Much appreciated.

Comment (juzhiyuan) [reply .github/workflows/backend-cli-test.yml:6]: got it, let's merge

**Review by juzhiyuan — APPROVED** [review]

File: .github/workflows/backend-cli-test.yml (hunk 1 above)
Comment (imjoey) [reply .github/workflows/backend-cli-test.yml:6]: > both the master and v2.x need to be checked IMO

 The codes in master branch will also be checked if we commit a PR targeted to `master` branch. So we can place different branch names here, respectively. Please correct me if I was wrong. Thanks.

**Comment by imjoey** [comment]
@juzhiyuan thank you very much for reviewing. Very glad that we can come to an agreement. Much appreciated for the openness. 👍

**Pull request merged**